server/lib/report-manager.js

```diff
@@ -38,8 +40,13 @@ class ReportManager
             throw new KError("alert's account does not exist");
         }
 
+        let defaultCurrency = await Config.byName('defaultCurrency').value;
+
         let operationsByAccount = new Map;
         for (let a of accounts) {
+            // We get the currency for this account, to format amounts correctly
+            let currency = a.currency ? a.currency : defaultCurrency;
+            a.currencyFormatter = formatCurrency(currency);
```

Comment:
It avoide code dupplication in         for (let account of accounts) loop and in             for (let pair of operationsByAccount.values()) one.